src/renderers/dom/shared/devtools/ReactDOMDuplicateSelectValuesDevtool.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactDOMDuplicateSelectValuesDevtool
+ */
+
+'use strict';
+
+var ReactComponentTreeDevtool = require('ReactComponentTreeDevtool');
+var warning = require('warning');
+var areEqual = require('areEqual');
+
+var didWarnDupeSelectValues = false;
+
+function handleElement(debugID, element) {
+  
+  if (didWarnDupeSelectValues) {
+    return;
+  }
+  
+  if (element == null) {
+    return;
+  }
+
+  if (element.type !== 'select') {
+    return;
+  }
+
+  if ((element.props == null) || (element.props.children == null)) {
+    return;
+  }
+
+  // Uncontrolled select elements can have duplicate values.
+  if (element.props.hasOwnProperty('defaultValue')) {
+    return;
+  }
+
+  let values = [];
+  let options = element.props.children;
+
+  //  If options is not iterable make it an array.
+  if (typeof options[Symbol.iterator] !== 'function') {
+    options = [options];
+  }
+
+  // Combine the values from all options into a single array.
+  for (const option of options) {
+    if (option.type === 'optGroup') {
+      if ((option.props != null) && (option.props.children != null)) {
+        let groupOptions = option.props.children;
+        //  If groupOptions is not iterable make it an array.
+        if (typeof groupOptions[Symbol.iterator] !== 'function') {
+          groupOptions = [groupOptions];
+        }
+        for (const groupOption of groupOptions) {
+          if ((groupOption.props != null) && (groupOption.props.value != null)) {
+            values.push(groupOption.props.value);
+          }
+        }
+      }
+    }
+
+    if (option.type === 'option') {
+      if ((option.props != null) && (option.props.value)) {
+        values.push(option.props.value);
+      }
+    }
+  }
+
+  if (values.length <= 1) {
+    return;
+  }
+
+  // Check the array for duplicate values.
+  for (var i = 0; i < values.length-1; i++) {
```

Comment:
This strategy for duplicate checking is a bit slow: O(n²)

We can check for duplicates in the first loop (line 52) by making `values` a dictionary and checking if the current value has already been seen:
``` js
const values = {};

for (const option of options) {
  ... // code for handling optgroups

  if (option.type === 'option'
    && option.props != null 
    && option.props.value) {
      const value = option.props.value;

      if (!values[value]) {
        values[value] = value;
      } else {
        // warn about duplicate
        return;
      }
  }
}
```

However, in reality, the chances of a `<select>` having enough `<options>`s to make this matter is small, but doing it this way does save us some lines of code.